### PR TITLE
Implemented custom idle timer logic

### LINKED ISSUE #1017: Remove auto locking setting from reader
Remove the "Allow device to sleep" setting from reader settings. Automatically disable sleeping for 25 minutes of no user interaction. After 25 minutes re-enable the system mechanism and let the device sleep.

### PR DESCRIPTION
**Pull request by michalrentka**
Closes #1017

### CODE REVIEW

File: Zotero/Controllers/IdleTimerController.swift [hunk 1]
```diff
@@ -8,35 +8,96 @@
...
     }
+
+    private func set(disabled: Bool) {
+        UIApplication.shared.isIdleTimerDisabled = disabled
+    }
+
+    private func forceStopIdleTimer() {
+        DDLogInfo("IdleTimerController: force stop timer")
+        activeProcesses = 0
+        stopCustomIdleTimer()
```
Comment (mvasilak) [inline Zotero/Controllers/IdleTimerController.swift:101]: This call will fail in the first guard of `stopCustomIdleTimer`

File: Zotero/Controllers/IdleTimerController.swift [hunk 2]
```diff
@@ -8,35 +8,96 @@
 
 import Foundation
 import UIKit
+
 import CocoaLumberjackSwift
+import RxSwift
 
 final class IdleTimerController {
+    private static let customIdleTimerTimemout = 1200
```
Comment (mvasilak) [inline Zotero/Controllers/IdleTimerController.swift:16]: `Timemeout` has a typo (2nd m). Also, why not make it directly
```suggestion
    private static let customIdleTimerTimeout: DispatchTimeInterval = .seconds(1200)
```

File: Zotero/Controllers/IdleTimerController.swift [hunk 3]
```diff
@@ -8,35 +8,96 @@
...
+            controller.activeTimer = timer
         }
     }
 
-    func enable() {
+    func stopCustomIdleTimer() {
         inMainThread { [weak self] in
-            guard let self = self else { return }
-            guard self.activeProcesses > 0 else {
+            guard let self, activeProcesses > 0 else {
```
Comment (mvasilak) [inline Zotero/Controllers/IdleTimerController.swift:72]: If `activeProcesses` is `0`, log a warning but continue to stop the timer.

File: Zotero/EventObservableWindow.swift [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+//  Created by Michal Rentka on 20.11.2024.
+//  Copyright © 2024 Corporation for Digital Scholarship. All rights reserved.
+//
+
+import UIKit
+
+final class EventObservableWindow: UIWindow {
+    override func sendEvent(_ event: UIEvent) {
+        super.sendEvent(event)
+        guard !(event.allTouches ?? []).isEmpty else { return }
```
Comment (mvasilak) [inline Zotero/EventObservableWindow.swift:14]: Are other events that we should consider, that may be applicable here, like device rotation?

File: Zotero/Controllers/IdleTimerController.swift (hunk 1 above)
Comment (michalrentka) [reply Zotero/Controllers/IdleTimerController.swift:104]: Fixed in b81703099fea811fc4dda794b99a9b6c4a36cc17

File: Zotero/Controllers/IdleTimerController.swift (hunk 2 above)
Comment (michalrentka) [reply Zotero/Controllers/IdleTimerController.swift]: Fixed in ec21879cebda856ecb862bc78281538db6705d31

File: Zotero/Controllers/IdleTimerController.swift (hunk 3 above)
Comment (michalrentka) [reply Zotero/Controllers/IdleTimerController.swift]: Fixed in ec21879cebda856ecb862bc78281538db6705d31

**Review by mvasilak — APPROVED** [review]

File: Zotero/EventObservableWindow.swift (hunk 4 above)
Comment (michalrentka) [reply Zotero/EventObservableWindow.swift:14]: Added in d4b36edc9e401ac3c1b89c67f1705ab1898c2593

**Pull request merged**